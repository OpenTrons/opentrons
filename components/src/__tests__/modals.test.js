// modal component tests
import React from 'react'
import Renderer from 'react-test-renderer'

import {Modal, ContinueModal, Overlay} from '..'

describe('modals', () => {
  test('Modal has a clickable overlay', () => {
    const onCloseClick = jest.fn()
    const root = Renderer.create(
      <Modal onCloseClick={onCloseClick}>
        children
      </Modal>
    ).root

    const overlay = root.findByProps({className: 'overlay'})
    overlay.props.onClick()

    expect(onCloseClick).toHaveBeenCalled()
  })

  test('ContinueModal has continue and cancel buttons', () => {
    const onCancelClick = jest.fn()
    const onContinueClick = jest.fn()
    const root = Renderer.create(
      <ContinueModal
        onCancelClick={onCancelClick}
        onContinueClick={onContinueClick}
      >
        children
      </ContinueModal>
    ).root

    const cancelButton = root.findByProps({title: 'Cancel'})
    const continueButton = root.findByProps({title: 'Continue'})

    cancelButton.props.onClick()
    expect(onCancelClick).toHaveBeenCalled()

    continueButton.props.onClick()
    expect(onContinueClick).toHaveBeenCalled()
  })

  test('ContinueModal calls onCancelClick on overlay click', () => {
    const onCancelClick = jest.fn()
    const onContinueClick = jest.fn()
    const root = Renderer.create(
      <ContinueModal
        onCancelClick={onCancelClick}
        onContinueClick={onContinueClick}
      >
        children
      </ContinueModal>
    ).root

    const overlay = root.findByProps({className: 'overlay'})

    overlay.props.onClick()
    expect(onCancelClick).toHaveBeenCalled()
  })

  test('Modal renders correctly', () => {
    const tree = Renderer.create(
      <Modal onCloseClick={() => {}} className='foo'>
        children
      </Modal>
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })

  test('ContinueModal renders correctly', () => {
    const tree = Renderer.create(
      <ContinueModal
        onCancelClick={() => {}}
        onContinueClick={() => {}}
      >
        children
      </ContinueModal>
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })

  test('Overlay renders correctly', () => {
    const tree = Renderer.create(
      <Overlay onClick={() => {}} />
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })
})
