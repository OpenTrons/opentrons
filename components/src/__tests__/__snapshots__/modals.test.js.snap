// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`modals AlertModal renders correctly 1`] = `
<div
  className="modal style"
>
  <div
    className="overlay clickable"
    onClick={[Function]}
  />
  <div
    className="modal_contents alert_modal_wrapper"
  >
    <div
      className="alert_modal_heading"
    >
      <svg
        aria-hidden="true"
        className="alert_modal_icon"
        fill="currentColor"
        height={undefined}
        style={undefined}
        version="1.1"
        viewBox="0 0 24 24"
        width={undefined}
        x={undefined}
        y={undefined}
      >
        <path
          d="M1 21h22L12 2 1 21zm12-3h-2v-2h2v2zm0-4h-2v-4h2v4z"
          fillRule="evenodd"
        />
      </svg>
      heading
    </div>
    <div
      className="alert_modal_contents"
    >
      children
    </div>
    <div
      className="alert_modal_buttons"
    >
      <button
        className="button_outline alert_button"
        disabled={undefined}
        onClick={[Function]}
        title={undefined}
        type="button"
      >
        a
      </button>
      <button
        className="button_outline alert_button"
        disabled={undefined}
        onClick={[Function]}
        title={undefined}
        type="button"
      >
        b
      </button>
    </div>
  </div>
</div>
`;

exports[`modals ContinueModal renders correctly 1`] = `
<div
  className="modal"
>
  <div
    className="overlay clickable"
    onClick={[Function]}
  />
  <div
    className="modal_contents alert_modal_wrapper no_alert_header"
  >
    <div
      className="alert_modal_contents"
    >
      children
    </div>
    <div
      className="alert_modal_buttons"
    >
      <button
        className="button_outline alert_button"
        disabled={undefined}
        onClick={[Function]}
        title="Cancel"
        type="button"
      >
        Cancel
      </button>
      <button
        className="button_outline alert_button"
        disabled={undefined}
        onClick={[Function]}
        title="Continue"
        type="button"
      >
        Continue
      </button>
    </div>
  </div>
</div>
`;

exports[`modals Modal renders correctly 1`] = `
<div
  className="modal foo"
>
  <div
    className="overlay clickable"
    onClick={[Function]}
  />
  <div
    className="modal_contents"
  >
    children
  </div>
</div>
`;

exports[`modals ModalPage renders correctly 1`] = `
<div
  className="modal"
>
  <div
    className="overlay"
    onClick={undefined}
  />
  <header
    className="title_bar title_bar"
  >
    <button
      className="button_flat inverted"
      disabled={undefined}
      onClick={[Function]}
      title="Back"
      type="button"
    >
      <svg
        aria-hidden="true"
        className="icon"
        fill="currentColor"
        height={undefined}
        style={undefined}
        version="1.1"
        viewBox="0 0 24 24"
        width={undefined}
        x={undefined}
        y={undefined}
      >
        <path
          d="M15.429 17.019l-4.242-4.59 4.242-4.59-1.312-1.41-5.545 6 5.545 6z"
          fillRule="evenodd"
        />
      </svg>
      back
    </button>
    <div
      className="title_wrapper"
    >
      <h1
        className="title"
      >
        Title
      </h1>
      <span
        className="separator"
      >
        |
      </span>
      <h2
        className="subtitle"
      >
        Subtitle
      </h2>
    </div>
  </header>
  <div
    className="modal_page_contents"
  >
    children
  </div>
</div>
`;

exports[`modals Overlay renders correctly 1`] = `
<div
  className="overlay clickable"
  onClick={[Function]}
/>
`;

exports[`modals SpinnerModalPage renders correctly 1`] = `
<div
  className="modal"
>
  <header
    className="title_bar title_bar"
  >
    <button
      className="button_flat inverted"
      disabled={true}
      onClick={undefined}
      title="Back"
      type="button"
    >
      <svg
        aria-hidden="true"
        className="icon"
        fill="currentColor"
        height={undefined}
        style={undefined}
        version="1.1"
        viewBox="0 0 24 24"
        width={undefined}
        x={undefined}
        y={undefined}
      >
        <path
          d="M15.429 17.019l-4.242-4.59 4.242-4.59-1.312-1.41-5.545 6 5.545 6z"
          fillRule="evenodd"
        />
      </svg>
      back
    </button>
    <div
      className="title_wrapper"
    >
      <h1
        className="title"
      >
        Title
      </h1>
      <span
        className="separator"
      >
        |
      </span>
      <h2
        className="subtitle"
      >
        Subtitle
      </h2>
    </div>
  </header>
  <div
    className="modal"
  >
    <div
      className="overlay"
      onClick={undefined}
    />
    <div
      className="spinner_modal_contents"
    >
      <svg
        aria-hidden="true"
        className="spinner_modal_icon spin"
        fill="currentColor"
        height={undefined}
        style={undefined}
        version="1.1"
        viewBox="0 0 512 512"
        width={undefined}
        x={undefined}
        y={undefined}
      >
        <path
          d="M304 48c0 26.51-21.49 48-48 48s-48-21.49-48-48 21.49-48 48-48 48 21.49 48 48zm-48 368c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48zm208-208c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48zM96 256c0-26.51-21.49-48-48-48S0 229.49 0 256s21.49 48 48 48 48-21.49 48-48zm12.922 99.078c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48c0-26.509-21.491-48-48-48zm294.156 0c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48c0-26.509-21.49-48-48-48zM108.922 60.922c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.491-48-48-48z"
          fillRule="evenodd"
        />
      </svg>
      <p />
    </div>
  </div>
</div>
`;
